### PR TITLE
Hd aditya Review updates, code refactor and Dashboard, calendar updates

### PR DESCRIPTION
**Pull request by gem-aditya-Saraswat**

### CODE REVIEW

File: src/test/java/selectors/CalenderXpath.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    public static String xpathStartEdge = "//span[@class='rdrStartEdge']";
+    public static By calenderStartRangeDate = By.xpath(xpathStartEdge);
+    public static String xpathEndEdge = "//span[contains(@class,'rdrEndEdge')]";
+    public static By calenderEndRangeDate = By.xpath(xpathEndEdge);
+    public static String xpathMonthOption = "//span[@class='rdrMonthPicker']/select/option";
+    public static By calenderMonthOption = By.xpath(xpathMonthOption);
+    public static String xpathWeekdayOption = "//span[@class='rdrWeekDay']";
+    public static String prevPrevious = "//div[@class='rdrMonthAndYearWrapper']//button[1]";
+    public static By prevPreviewBtn = By.xpath(prevPrevious);
+    public static String prevNext = "//div[@class='rdrMonthAndYearWrapper']//button[2]";
```
Comment (gem-akhil-singh) [inline src/test/java/selectors/CalenderXpath.java:25]: Does Indexing is the last thing which could be done here ?  If not , then try creating xpath independent of index

File: src/test/java/stepdefination/DashboardEmployeeViewStepDefinition.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            logger.info("An exception occurred!", e);
+            GemTestReporter.addTestStep("ERROR", "SOME ERROR OCCURRED" + e, STATUS.FAIL);
+        }
+    }
+
+    @Then("^Verify New tab open$")
+    public void newTabVerify(){
+        try {
+            prevURL = getCurrentURL();
+            List<String> windowList = new ArrayList<>(getWindowHandles());
```
Comment (gem-akhil-singh) [inline src/test/java/stepdefination/DashboardEmployeeViewStepDefinition.java:35]: Dont Use List for storing window handels Use Set for storing window handles. List can have duplicate values Set can't;
Read Here For Detail ; https://stackoverflow.com/questions/36053853/why-we-use-set-in-window-handles-in-selenium-and-not-list

File: src/test/java/stepdefination/DashboardEmployeeViewStepDefinition.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            logger.info("An exception occurred!", e);
+            GemTestReporter.addTestStep("ERROR", "SOME ERROR OCCURRED" + e, STATUS.FAIL);
+        }
+    }
+
+    @Then("^Verify user logged out successfully$")
+    public void logoutVerification(){
+        try{
+            waitSec(3);
+            String expectedUrl = "https://helpdeskui-np.geminisolutions.com/#/";
```
Comment (gem-akhil-singh) [inline src/test/java/stepdefination/DashboardEmployeeViewStepDefinition.java:157]: These Properties should be fetched from resource files, Create a property or json file in Resources and fetch these values from there

Comment (gem-akhil-singh) [reply src/test/java/stepdefination/DashboardEmployeeViewStepDefinition.java:157]: expected URL

File: src/test/java/utilities/CalenderUtils.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        }catch (Exception e){
+            logger.info("An exception occurred!", e);
+            GemTestReporter.addTestStep("ERROR", "SOME ERROR OCCURRED" + e, STATUS.FAIL);
+        }
+    }
+    public void verifySelectedMaximumDateRange(){
+        try{
+            List<WebElement> inrangeDates = getElements(CalenderXpath.calenderInRangeDates);
+            if(isExist(CalenderXpath.calenderStartRangeDate) && isExist(CalenderXpath.calenderEndRangeDate)){
+                if (inrangeDates.size() == 29 || inrangeDates.size() == 28 || inrangeDates.size() == 26 || inrangeDates.size() == 25){
```
Comment (gem-akhil-singh) [inline src/test/java/utilities/CalenderUtils.java:36]: I have Recommended changes in the Last PR for the Dates its not yet Imeplemented

File: src/test/java/utilities/CalenderUtils.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            }
+        }catch(Exception e){
+            logger.info("An exception occurred!", e);
+            GemTestReporter.addTestStep("ERROR", "SOME ERROR OCCURRED" + e, STATUS.FAIL);
+        }
+    }
+    
+    public void verifyDataForRange(String startDate, String endDate){
+        try {
+            ArrayList<String> dateCreateOn = new ArrayList<>();
```
Comment (gem-akhil-singh) [inline src/test/java/utilities/CalenderUtils.java:108]: What is the Use of this ArrayList here ? its not being called or added or fetched values in the code

File: src/test/java/utilities/LoginUtils.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+
+public class LoginUtils {
+    final Logger logger = LoggerFactory.getLogger(LoginStepDefinition.class);
+    public void verifyLogin(){
+        try {
+            waitSec(5);
+            String curURL = getCurrentURL();
+            GemTestReporter.addTestStep("Current URL", "Current URL : "+ curURL, STATUS.INFO, takeSnapShot());
+            String expURL = "https://helpdeskui-np.geminisolutions.com/#/dashboard/0";
```
Comment (gem-akhil-singh) [inline src/test/java/utilities/LoginUtils.java:19]: Fetch expURL from Resources folder, create a resource File

Comment (gem-akhil-singh) [reply src/test/java/utilities/LoginUtils.java:19]: or can also be used from gemjar config file , create a key and use

**Review by gem-akhil-singh — CHANGES_REQUESTED** [review]
Please address the mentioned changes in the respective files and code blocks.

File: src/test/java/selectors/CalenderXpath.java (hunk 1 above)
Comment (gem-aditya-Saraswat) [reply src/test/java/selectors/CalenderXpath.java:25]: Yes, I have tried. But indexing is the last approach here.

File: src/test/java/utilities/CalenderUtils.java (hunk 5 above)
Comment (gem-aditya-Saraswat) [reply src/test/java/utilities/CalenderUtils.java:108]: I am working in this method it is not yet completed.

File: src/test/java/utilities/LoginUtils.java (hunk 6 above)
Comment (gem-aditya-Saraswat) [reply src/test/java/utilities/LoginUtils.java:19]: Will look into the right approach for this.

**Pull request closed without merging**